mp4parse/src/lib.rs

```diff
@@ -1163,19 +1178,45 @@ fn read_esds<T: Read>(src: &mut BMFFBox<T>) -> Result<ES_Descriptor> {
         }
 
         // find DecoderConfig descriptor (tag = DECODER_CONFIG_TAG)
-        let dcds = try!(esds.read_u8());
-        if dcds != DECODER_CONFIG_TAG {
-            return Err(Error::Unsupported("fail to parse decoder config descriptor"));
+        if esds_end > esds.position() {
+            next_tag = try!(esds.read_u8());
```

Comment:
Since these uses of `next_tag` start with a fresh assignment within a block, please drop the `mut` keyword from the outer declaration and use a fresh `let` here and below. Rust will not warn about the shadowing, and using a local in each block reduces the amount of code which must be understood at once.